receiver/azuremonitorreceiver: fix lint issues

```
factory.go:47:28: unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _ (revive)
func createMetricsReceiver(ctx context.Context, params receiver.CreateSettings, rConf component.Config, consumer consumer.Metrics) (receiver.Metrics, error) {
                           ^
scraper.go:135:30: unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _ (revive)
func (s *azureScraper) start(ctx context.Context, host component.Host) (err error) {
                             ^
scraper_test.go:105:40: unused-parameter: parameter 'options' seems to be unused, consider removing or renaming it as _ (revive)
func (acm *armClientMock) NewListPager(options *armresources.ClientListOptions) *runtime.Pager[armresources.ClientListResponse] {
                                       ^
scraper_test.go:123:76: unused-parameter: parameter 'options' seems to be unused, consider removing or renaming it as _ (revive)
func (mdcm *metricsDefinitionsClientMock) NewListPager(resourceURI string, options *armmonitor.MetricDefinitionsClientListOptions) *runtime.Pager[armmonitor.MetricDefinitionsClientListResponse] {
                                                                           ^
scraper_test.go:140:42: unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _ (revive)
func (mvcm metricsValuesClientMock) List(ctx context.Context, resourceURI string, options *armmonitor.MetricsClientListOptions) (armmonitor.MetricsClientListResponse, error) {
```

--- a/receiver/azuremonitorreceiver/factory.go
+++ b/receiver/azuremonitorreceiver/factory.go
@@ -44,7 +44,7 @@ func createDefaultConfig() component.Config {
 	}
 }
 
-func createMetricsReceiver(ctx context.Context, params receiver.CreateSettings, rConf component.Config, consumer consumer.Metrics) (receiver.Metrics, error) {
+func createMetricsReceiver(_ context.Context, params receiver.CreateSettings, rConf component.Config, consumer consumer.Metrics) (receiver.Metrics, error) {
 	cfg, ok := rConf.(*Config)
 	if !ok {
 		return nil, errConfigNotAzureMonitor

--- a/receiver/azuremonitorreceiver/scraper.go
+++ b/receiver/azuremonitorreceiver/scraper.go
@@ -132,7 +132,7 @@ func (s *azureScraper) GetMetricsValuesClient() MetricsValuesClient {
 	return client
 }
 
-func (s *azureScraper) start(ctx context.Context, host component.Host) (err error) {
+func (s *azureScraper) start(_ context.Context, host component.Host) (err error) {
 	s.cred, err = s.azIDCredentialsFunc(s.cfg.TenantID, s.cfg.ClientID, s.cfg.ClientSecret, nil)
 	if err != nil {
 		return err

--- a/receiver/azuremonitorreceiver/scraper_test.go
+++ b/receiver/azuremonitorreceiver/scraper_test.go
@@ -137,7 +137,7 @@ type metricsValuesClientMock struct {
 	lists map[string]map[string]armmonitor.MetricsClientListResponse
 }
 
-func (mvcm metricsValuesClientMock) List(ctx context.Context, resourceURI string, _ *armmonitor.MetricsClientListOptions) (armmonitor.MetricsClientListResponse, error) {
+func (mvcm metricsValuesClientMock) List(ctx context.Context, resourceURI string, options *armmonitor.MetricsClientListOptions) (armmonitor.MetricsClientListResponse, error) {
 	return mvcm.lists[resourceURI][*options.Metricnames], nil
 }
 

--- a/receiver/azuremonitorreceiver/scraper_test.go
+++ b/receiver/azuremonitorreceiver/scraper_test.go
@@ -102,7 +102,7 @@ type armClientMock struct {
 	pages   []armresources.ClientListResponse
 }
 
-func (acm *armClientMock) NewListPager(options *armresources.ClientListOptions) *runtime.Pager[armresources.ClientListResponse] {
+func (acm *armClientMock) NewListPager(_ *armresources.ClientListOptions) *runtime.Pager[armresources.ClientListResponse] {
 	return runtime.NewPager(runtime.PagingHandler[armresources.ClientListResponse]{
 		More: func(page armresources.ClientListResponse) bool {
 			return acm.current < len(acm.pages)
@@ -120,7 +120,7 @@ type metricsDefinitionsClientMock struct {
 	pages   map[string][]armmonitor.MetricDefinitionsClientListResponse
 }
 
-func (mdcm *metricsDefinitionsClientMock) NewListPager(resourceURI string, options *armmonitor.MetricDefinitionsClientListOptions) *runtime.Pager[armmonitor.MetricDefinitionsClientListResponse] {
+func (mdcm *metricsDefinitionsClientMock) NewListPager(resourceURI string, _ *armmonitor.MetricDefinitionsClientListOptions) *runtime.Pager[armmonitor.MetricDefinitionsClientListResponse] {
 	return runtime.NewPager(runtime.PagingHandler[armmonitor.MetricDefinitionsClientListResponse]{
 		More: func(page armmonitor.MetricDefinitionsClientListResponse) bool {
 			return mdcm.current[resourceURI] < len(mdcm.pages[resourceURI])
@@ -137,7 +137,7 @@ type metricsValuesClientMock struct {
 	lists map[string]map[string]armmonitor.MetricsClientListResponse
 }
 
-func (mvcm metricsValuesClientMock) List(ctx context.Context, resourceURI string, options *armmonitor.MetricsClientListOptions) (armmonitor.MetricsClientListResponse, error) {
+func (mvcm metricsValuesClientMock) List(ctx context.Context, resourceURI string, _ *armmonitor.MetricsClientListOptions) (armmonitor.MetricsClientListResponse, error) {
 	return mvcm.lists[resourceURI][*options.Metricnames], nil
 }
 

--- a/receiver/bigipreceiver/config_test.go
+++ b/receiver/bigipreceiver/config_test.go
@@ -38,7 +38,7 @@ func TestValidate(t *testing.T) {
 			expectedErr: multierr.Combine(
 				errMissingUsername,
 				errMissingPassword,
-				fmt.Errorf("%s: %w", errInvalidEndpoint, errors.New(`parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`)),
+				fmt.Errorf("%w: %w", errInvalidEndpoint, errors.New(`parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`)),
 			),
 		},
 		{
@@ -51,7 +51,7 @@ func TestValidate(t *testing.T) {
 			},
 			expectedErr: multierr.Combine(
 				errMissingPassword,
-				fmt.Errorf("%s: %w", errInvalidEndpoint, errors.New(`parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`)),
+				fmt.Errorf("%w: %w", errInvalidEndpoint, errors.New(`parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`)),
 			),
 		},
 		{
@@ -64,7 +64,7 @@ func TestValidate(t *testing.T) {
 			},
 			expectedErr: multierr.Combine(
 				errMissingUsername,
-				fmt.Errorf("%s: %w", errInvalidEndpoint, errors.New(`parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`)),
+				fmt.Errorf("%w: %w", errInvalidEndpoint, errors.New(`parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`)),
 			),
 		},
 		{
@@ -77,7 +77,7 @@ func TestValidate(t *testing.T) {
 				},
 			},
 			expectedErr: multierr.Combine(
-				fmt.Errorf("%s: %w", errInvalidEndpoint, errors.New(`parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`)),
+				fmt.Errorf("%w: %w", errInvalidEndpoint, errors.New(`parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`)),
 			),
 		},
 		{

--- a/receiver/bigipreceiver/config_test.go
+++ b/receiver/bigipreceiver/config_test.go
@@ -4,7 +4,6 @@
 package bigipreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/bigipreceiver"
 
 import (
-	"errors"
 	"fmt"
 	"path/filepath"
 	"testing"
@@ -38,7 +37,7 @@ func TestValidate(t *testing.T) {
 			expectedErr: multierr.Combine(
 				errMissingUsername,
 				errMissingPassword,
-				fmt.Errorf("%w: %w", errInvalidEndpoint, errors.New(`parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`)),
+				fmt.Errorf("%w: %s", errInvalidEndpoint, `parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`),
 			),
 		},
 		{
@@ -51,7 +50,7 @@ func TestValidate(t *testing.T) {
 			},
 			expectedErr: multierr.Combine(
 				errMissingPassword,
-				fmt.Errorf("%w: %w", errInvalidEndpoint, errors.New(`parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`)),
+				fmt.Errorf("%w: %s", errInvalidEndpoint, `parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`),
 			),
 		},
 		{
@@ -64,7 +63,7 @@ func TestValidate(t *testing.T) {
 			},
 			expectedErr: multierr.Combine(
 				errMissingUsername,
-				fmt.Errorf("%w: %w", errInvalidEndpoint, errors.New(`parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`)),
+				fmt.Errorf("%w: %s", errInvalidEndpoint, `parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`),
 			),
 		},
 		{
@@ -77,7 +76,7 @@ func TestValidate(t *testing.T) {
 				},
 			},
 			expectedErr: multierr.Combine(
-				fmt.Errorf("%w: %w", errInvalidEndpoint, errors.New(`parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`)),
+				fmt.Errorf("%w: %s", errInvalidEndpoint, `parse "invalid://endpoint:  12efg": invalid port ":  12efg" after host`),
 			),
 		},
 		{

--- a/receiver/carbonreceiver/transport/mock_reporter.go
+++ b/receiver/carbonreceiver/transport/mock_reporter.go
@@ -27,14 +27,14 @@ func (m *MockReporter) OnDataReceived(ctx context.Context) context.Context {
 	return ctx
 }
 
-func (m *MockReporter) OnTranslationError(ctx context.Context, err error) {
+func (m *MockReporter) OnTranslationError(_ context.Context, err error) {
 }
 
-func (m *MockReporter) OnMetricsProcessed(ctx context.Context, numReceivedMetricPoints int, err error) {
+func (m *MockReporter) OnMetricsProcessed(_ context.Context, numReceivedMetricPoints int, err error) {
 	m.wgMetricsProcessed.Done()
 }
 
-func (m *MockReporter) OnDebugf(template string, args ...interface{}) {
+func (m *MockReporter) OnDebugf(_ string, args ...interface{}) {
 }
 
 // WaitAllOnMetricsProcessedCalls blocks until the number of expected calls

--- a/receiver/collectdreceiver/collectd.go
+++ b/receiver/collectdreceiver/collectd.go
@@ -160,14 +160,14 @@ func (r *collectDRecord) newPoint(val *json.Number) (*metricspb.Point, bool, err
 // becomes a dimension.
 func (r *collectDRecord) getReasonableMetricName(index int, attrs map[string]string) (string, bool) {
 	usedDsName := false
-	cap := 0
+	capacity := 0
 	if r.TypeS != nil {
-		cap += len(*r.TypeS)
+		capacity += len(*r.TypeS)
 	}
 	if r.TypeInstance != nil {
-		cap += len(*r.TypeInstance)
+		capacity += len(*r.TypeInstance)
 	}
-	parts := make([]byte, 0, cap)
+	parts := make([]byte, 0, capacity)
 
 	if !isNilOrEmpty(r.TypeS) {
 		parts = append(parts, *r.TypeS...)

